libs/postgres_ffi/src/xlog_utils.rs

```diff
@@ -154,9 +155,11 @@ fn find_end_of_wal_segment(
     file.seek(SeekFrom::Start(offs as u64))?;
     let mut rec_hdr = [0u8; XLOG_RECORD_CRC_OFFS];
 
+    trace!("find_end_of_wal_segment(data_dir={}, segno={}, tli={}, wal_seg_size={}, start_offset=0x{:x})", data_dir.display(), segno, tli, wal_seg_size, start_offset);
```

Comment:
Should we validate here that 
1. start_offset is < wal_seg_size?
2. wal_seg_size is a power of 2 and between 1Mb and 1Gb?